interchain-token-factory/src/proxy.rs

```diff
@@ -0,0 +1,236 @@
+multiversx_sc::imports!();
+
+use crate::constants::{Hash, ManagedBufferAscii, TokenId};
+use core::ops::Deref;
+use multiversx_sc::api::KECCAK256_RESULT_LEN;
+
+use interchain_token_service::proxy::ProxyTrait as _;
+use interchain_token_service::ProxyTrait as _;
+use operatable::ProxyTrait as _;
+use token_manager::constants::TokenManagerType;
+use token_manager::minter::ProxyTrait as _;
+use token_manager::ProxyTrait as _;
+
+#[multiversx_sc::module]
+pub trait ProxyModule {
+    fn its_chain_name_hash(&self) -> ManagedByteArray<KECCAK256_RESULT_LEN> {
+        self.interchain_token_service_proxy(self.interchain_token_service().get())
+            .chain_name_hash()
+            .execute_on_dest_context()
+    }
+
+    fn its_interchain_token_id(
+        &self,
+        sender: &ManagedAddress,
+        salt: &Hash<Self::Api>,
+    ) -> TokenId<Self::Api> {
+        self.interchain_token_service_proxy(self.interchain_token_service().get())
```

Comment:
This is a keccak256 hash based on the sender and salt so it can have indefinite values and is used only once or twice by this contract, since the endpoints calling this are used only once for a sender/salt combination.

Since this is a proxy contract that exists for easier integration with the InterchainTokenService, I think it is fine how it is right now (the same approach is also used for EVM).